### PR TITLE
Move GetImageMoniker() Glyph extension method from Editor.wpf to Edit…

### PR DESCRIPTION
**Pull request by JieCarolHu**
…orFeatures

<details><summary>Ask Mode template not completed</summary>

### Customer scenario

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

### Bugs this fixes

(either VSO or GitHub links)

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

File: src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs [hunk 1]
```diff
@@ -1,12 +1,206 @@
...
 {
     internal static class GlyphExtensions
     {
+        public static ImageId GetImageId(this Glyph glyph)
+        {
+            var moniker = glyph.GetImageMoniker();
+            return new ImageId(moniker.Guid, moniker.Id);
+        }
+
+        public static ImageMoniker GetImageMoniker(this Glyph glyph)
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs]: I thought part of the goal was to not have any mention of `ImageMoniker` in EditorFeatures, and to instead directly translate from Glyph to ImageId without the middle step (for Glyph.Assembly, it'd look like `new ImageId(KnownImageIds.ImageCatalogGuid, KnownImageIds.Assembly)`)

Comment (dpoeschl) [reply src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs]: (And leave the ImageMoniker one in EditorFeatures.WPF for consumers who need actual ImageMonikers)

Comment (jasonmalinowski) [reply src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs]: Correct: this needs to become GetImageId() with the really huge switch. Then GetImageMoniker() stays in EditorFeatures.Wpf and just converts from Id to Moniker.

File: src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs [hunk 2]
```diff
@@ -1,12 +1,201 @@
...
+                case Glyph.StatusInformation:
+                    return new ImageId(KnownMonikers.StatusInformation.Guid, KnownMonikers.StatusInformation.Id);
+
+                case Glyph.NuGet:
+                    return new ImageId(KnownMonikers.NuGet.Guid, KnownMonikers.NuGet.Id);
+
+                default:
+                    throw new ArgumentException("glyph");
+            }
+        }
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs]: Nit: The team typically puts a blank line between this close curly brace and the start of the next method.

File: src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs [hunk 3]
```diff
@@ -1,12 +1,202 @@
...
+            // VS for mac cannot refer to ImageMoniker
+            // so we need to expose ImageId instead of ImageMoniker here
+            // and expose ImageMoniker in the EditorFeatures.wpf.dll
+            switch (glyph)
+            {
+                case Glyph.None:
+                    return default;
+
+                case Glyph.Assembly:
+                    return new ImageId(KnownMonikers.Assembly.Guid, KnownMonikers.Assembly.Id);
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs]: We have to be careful here: if this is embedding constants we're OK; if this is embedding references to the KnownMonikers type then that might cause mono to load ImageMoniker and thus still explode. Can we use KnownImageIds instead?

File: src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs [hunk 4]
```diff
@@ -1,12 +1,202 @@
...
+                    return new ImageId(KnownMonikers.IntellisenseWarning.Guid, KnownMonikers.IntellisenseWarning.Id);
+
+                case Glyph.StatusInformation:
+                    return new ImageId(KnownMonikers.StatusInformation.Guid, KnownMonikers.StatusInformation.Id);
+
+                case Glyph.NuGet:
+                    return new ImageId(KnownMonikers.NuGet.Guid, KnownMonikers.NuGet.Id);
+
+                default:
+                    throw new ArgumentException("glyph");
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs]: nameof(glyph)

**Comment by JieCarolHu** [comment]
@Pilchie Sorry I merged it without asking for ask mode sign off. Could you please approve it. Thanks

**Comment by Pilchie** [comment]
Retractively approved for 15.7 P2.

**Pull request merged**